enhancements/network/user-defined-network-segmentation.md

```diff
@@ -0,0 +1,955 @@
+ - Indicate this network will be the primary network
+
+After creating this CRD, I can check the status of the CRD to ensure it is actively being used as the primary network
+for my namespace by OVN-Kubernetes. Once verified, I can now create pods and they will be in their own isolated SDN.
+
+#### Admin Use Case
+
+As an admin, I have a customer who has multiple namespaces and wants to connect them all to the same private network. In
+order to accomplish this, I first create an admin network CRD that is cluster scoped and indicate:
+
+- Type of network (Layer 3 or Layer 2)
+- IP addressing scheme I wish to use (optional)
+- Indicate this network will be the primary network
+- Selector to decide which namespaces may connect to this network. May use the ```kubernetes.io/metadata.name``` label to 
+guarantee uniqueness and eliminates the ability to falsify access.
+
+After creating the CRD, check the status to ensure OVN-Kubernetes has accepted this network to serve the namespaces
+selected. Now tenants may go ahead and be provisioned their namespace.
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+The management cluster should have no reason to use multiple networks, unless it for security reasons it makes sense
+to use native network isolation over network policy. It makes more sense that multiple primary networks will be used in
+the hosted cluster in order to provide tenants with better isolation from each other, without the need for network
+policy. There should be no hypershift platform-specific considerations with this feature.
+
+#### Standalone Clusters
+
+Full support.
+
+#### Single-node Deployments or MicroShift
+
+SNO and Microshift will both have full support for creating multiple primary networks. There may be some increased
+resource usage when scaling up number of networks as it requires more configuration in OVN, and more processing in
+OVN-Kubernetes.
+
+### Risks and Mitigations
+
+The biggest risk with this feature is hitting scale limitations. With many namespaces and networks, the number of
+internal OVN objects will multiply, as well as internal kernel devices, rules, VRFs. There will need to be a large-scale
+effort to determine how many networks we can comfortably support. 
+
+There is also a risk of breaking secondary projects that integrate with OVN-Kubernetes, such as Metal LB or Submariner.
+
+### Drawbacks
+
+As described in the Design Details section, this proposal will require reserving two IPs per network in the masquerade
+subnet. This is a private subnet only used internally by OVN-Kubernetes, but it will require increasing the subnet size
+in order to accommodate multiple networks. Today this subnet by default is configured as a /29 for IPv4, and only 6 IP
+addresses are used. With this new design, users will need to reconfigure their subnet to be large enough to hold the
+desired number of networks. Note, it is already supported  to change the masquerade subnet post-installation.
```

Comment:
If we we talking about V4MasqueradeSubnet here, its not supported to change post installation.

Reply:
Ah you are right....
https://github.com/openshift/api/pull/1626/commits/6c48eeb34745e59ab3dd50bd260ba9489b7a0ad2#diff-2e98c189fdf696589c6d28124557b9cf96a329e82f790b4e61209351cc68648dR462

@tssurya we will need to fix this. Thanks for pointing that out @anuragthehatter